src/main/java/frc/trigon/robot/utilities/mechanisms/SingleJointedArmMechanism2d.java

```diff
@@ -0,0 +1,89 @@
+package frc.trigon.robot.utilities.mechanisms;
+
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import org.littletonrobotics.junction.Logger;
+
+/**
+ * A Mechanism2d object to display the current angle and the target angle of a single jointed arm.
+ */
+public class SingleJointedArmMechanism2d {
+    private final String key;
+    private final Mechanism2d mechanism;
+    private final MechanismLigament2d
+            currentPositionLigament,
+            targetPositionLigament;
+
+    public SingleJointedArmMechanism2d(String key) {
+        this(key, MechanismConstants.BLUE);
+    }
+
+    public SingleJointedArmMechanism2d(String key, Color8Bit mechanismColor) {
+        this.key = key;
+        this.mechanism = new Mechanism2d(2 * MechanismConstants.MECHANISM_LINE_LENGTH, 2 * MechanismConstants.MECHANISM_LINE_LENGTH);
+        MechanismRoot2d root = mechanism.getRoot("AngleRoot", MechanismConstants.MECHANISM_LINE_LENGTH, MechanismConstants.MECHANISM_LINE_LENGTH);
+        this.currentPositionLigament = root.append(new MechanismLigament2d("ZCurrentPositionLigament", MechanismConstants.MECHANISM_LINE_LENGTH, 0, MechanismConstants.MECHANISM_LINE_WIDTH, mechanismColor));
+        this.targetPositionLigament = root.append(new MechanismLigament2d("TargetPositionLigament", MechanismConstants.MECHANISM_LINE_LENGTH, 0, MechanismConstants.MECHANISM_LINE_WIDTH, MechanismConstants.GRAY));
```

Comment:
make sure that its whole mechanism size isn't exactly the length of the arm; give some wiggle room as a 10% constant or something. So the mechanism will be 10% longer than the arm's length together * 2.